hal/inc/pinmap_hal.h

```diff
@@ -37,24 +26,24 @@
 extern "C" {
 #endif
 
-/* Exported types ------------------------------------------------------------*/
+typedef struct Hal_Pin_Info Hal_Pin_Info;
```

Comment:
Let's remove this and rely on `pinmap_impl.h` to define `Hal_Pin_Info` for us.

Reply:
This a forward declaration, `Hal_Pin_Info` will be used in driver files or wiring files, they only include `pinmap_hal.h`, we leave the detailed structure implementation in the different platform folders.